### ISSUE TITLE
Unclear wording in “Tools for Monitoring Resources”

### ISSUE THREAD

**Issue by schelv** [kubernetes/website#31483]
Hi!

I have a question about the [resource usage monitoring](https://kubernetes.io/docs/tasks/debug-application-cluster/resource-usage-monitoring/) page.

In the section "Full metrics pipeline" it says: 

> Prometheus, a CNCF project, can natively monitor Kubernetes, nodes, and Prometheus itself. Full metrics pipeline projects that are not part of the CNCF are outside the scope of Kubernetes documentation.

What does the first sentence mean?

What information can prometheus monitor? is this full metrics? 
And how can I use prometheus to access this information? 
The word "natively" suggest this is easy.

**Comment by sftim** [comment]
I think this is mainly a request for help @schelv 

You might like to ask for more advice about Prometheus in https://discuss.kubernetes.io/, or search the web for a guide on how to monitor Kubernetes with Prometheus (eg https://www.google.com/search?q=monitor+kubernetes+with+prometheus). There are some good guides out there.

/kind support

BTW The word natively does not mean “in a way that is simple to learn”; it means “without requiring additional components”.

**Comment by schelv** [comment]
@sftim 
Thank you for the quick reply.

If you don't have the knowledge yet about a topic, it is difficult to estimate the quality of a guide.
Most of the time the documentation is more accurate and up-to-date.
It would be nice if the documentation would be understandable for a beginner.
That is what this issue is about.

This:
> Prometheus, a CNCF project, can natively monitor Kubernetes, nodes, and Prometheus itself. Full metrics pipeline projects that are not part of the CNCF are outside the scope of Kubernetes documentation.

Is the same as this (right?):
> Prometheus, a CNCF project, can monitor Kubernetes and nodes without requiring additional components. Full metrics pipeline projects that are not part of the CNCF are outside the scope of Kubernetes documentation.

That means that the current documentation is saying: "Prometheus is a full metrics pipeline that does not require additional components."

Is this information really correct?
[One of the guides from google](https://sysdig.com/blog/kubernetes-monitoring-prometheus/) says I will need to install kube-state-metrics as well.

**Comment by sftim** [comment]
These two paragraphs are not equivalent:

> Prometheus, a CNCF project, can natively monitor Kubernetes, nodes, and Prometheus itself. Full metrics pipeline projects that are not part of the CNCF are outside the scope of Kubernetes documentation.

> Prometheus, a CNCF project, can monitor Kubernetes and nodes without requiring additional components. Full metrics pipeline projects that are not part of the CNCF are outside the scope of Kubernetes documentation.

We should improve that page to clarify the situation. The existing text is not clear enough.

/kind bug
/language en
/remove-kind support
/retitle Unclear wording in “Tools for Monitoring Resources”
/priority backlog
/triage accepted

**Comment by subhamgcon** [comment]
Hey, @mehabhalodiya @sftim I want to work on this issue. Can you please help me with how to get started with it?

**Comment by subhamgcon** [comment]
I want to know what kind of description you are looking for.

**Comment by sftim** [comment]
We want to change the paragraph:

> Prometheus, a CNCF project, can natively monitor Kubernetes, nodes, and Prometheus itself. Full metrics pipeline projects that are not part of the CNCF are outside the scope of Kubernetes documentation.

The changes we should make are to emphasize the second sentence more (maybe more that part first). We should mention [OpenMetrics](https://openmetrics.io/) and make it clear that Kubernetes is designed to work with OpenMetrics.
Instead of mentioning Prometheus at all, I recommend that we hyperlink to the CNCF Landscape, and specifically to https://landscape.cncf.io/card-mode?category=monitoring&project=graduated,incubating,member,no&grouping=category&sort=stars

We should explain along with the link that the Kubernetes project doesn't provide a recommendation of projects (it's up to the reader to pick). The overall point we want to make is that if someone wants a full metrics pipeline, they need to design and deploy their own infrastructure platform of which Kubernetes is only one part.

@subhamgcon does that give you an idea about some new wording?

**Comment by subhamgcon** [comment]
@sftim Yes, you suggested rephrasing the sentence so that the user lands on the CNCF landscape page and so that it is cleared the Kubernetes doesn’t recommend using any project at all, it is up to the user.  Am I right?

**Comment by NitishKumar06** [comment]
@sftim  If the purpose is to make newbies understand, what's the point of mentioning a link? I think we can remove the first line (if you say else not required). And talking about the second line, we can add a new line mentioning something like _if someone wants a full metrics pipeline, they need to design and deploy their own infrastructure platform of which Kubernetes is only one part._ WDYT @sftim ?

**Comment by sftim** [comment]
> @sftim If the purpose is to make newbies understand, what's the point of mentioning a link?

Novices can and do follow hyperlinks, provided we give folk enough context to know why they'd want to.

**Comment by GitBatra** [comment]
Hey @sftim @mehabhalodiya On this issue i want to work upon. Can you please help me with how to begin with?

**Comment by sftim** [comment]
@GitBatra (and anyone else looking for help) a good place to ask for that advice is the #sig-docs channel in the [Kubernetes Slack organization](https://github.com/kubernetes/community/blob/master/communication/README.md#slack).

**Comment by sftim** [comment]
https://github.com/kubernetes/website/issues/31483#issuecomment-1148521363 should cover what changes to make. It's OK to ask questions here - on GitHub - if that outline isn't clear. Please try to be specific about what part of that advice needs explaining further.

**Comment by sftim** [comment]
> @sftim Yes, you suggested rephrasing the sentence so that the user lands on the CNCF landscape page and so that it is cleared the Kubernetes doesn’t recommend using any project at all, it is up to the user. Am I right?

You are right @subhamgcon . Kubernetes does not recommend one metrics / monitoring product or project over any other.

**Comment by Aut0R3V** [comment]
I think I can work on this. Can I assign this to myself @NitishKumar06

**Comment by NitishKumar06** [comment]
@Aut0R3V Please start working on this issue. Let me know if you need any help from my side.

**Comment by ShivangShandilya** [comment]
@sftim I'm new to Kubernetes but I still gave this issue a try and changed the paragraph, which according to me will help people understand it a little better, so should I share it here first and you can review it? Regards

**Comment by ashutosh887** [comment]
@divya-mohan0209 Let me work on this

**Comment by NitishKumar06** [comment]
> @divya-mohan0209 Let me work on this

A [valid PR](https://github.com/kubernetes/website/pull/38816) has already been raised for this issue which will close this issue once that PR is merged.

**Comment by anubhav1206** [comment]
Can i work on this issue?

**Comment by NitishKumar06** [comment]
> > @divya-mohan0209 Let me work on this
> 
> A [valid PR](https://github.com/kubernetes/website/pull/38816) has already been raised for this issue which will close this issue once that PR is merged.

@anubhav1206

**Comment by sftim** [comment]
/unassign @Swarnlataaa 

There's no need to work on closed issues.
